File: 200-number-of-islands/200-number-of-islands.java
class Solution {
    void removeLand(char[][] grid, int i, int j) {
        if(i < 0 || j < 0 || i >= grid.length || j >= grid[0].length || grid[i][j] == '0')
            return;
        grid[i][j] = '0';
        removeLand(grid, i - 1, j);
        removeLand(grid, i + 1, j);
        removeLand(grid, i, j - 1);
        removeLand(grid, i, j + 1);
    }
    public int numIslands(char[][] grid) {
        int islands = 0;
        for(int i = 0; i < grid.length; ++i) {
            for(int j = 0; j < grid[0].length; ++j) {
                if(grid[i][j] == '1') {
                    removeLand(grid, i, j);
                    ++islands;
                }
            }
        }
        return islands;
    }
}

File: 200-number-of-islands/200-number-of-islands.py
class Solution:
    def removeLand(self, grid: List[List[str]], i: int, j: int) -> None:
        if i == -1 or i == len(grid) or j == -1 or j == len(grid[0])  or grid[i][j] == '0':
            return
        grid[i][j] = '0'
        self.removeLand(grid, i - 1, j)
        self.removeLand(grid, i, j + 1)
        self.removeLand(grid, i + 1, j)
        self.removeLand(grid, i, j - 1)
        
    def numIslands(self, grid: List[List[str]]) -> int:
        islands = 0
        for i in range (len(grid)):
            for j in range (len(grid[0])):
                if grid[i][j] == '1':
                    self.removeLand(grid, i, j)
                    islands += 1
        return islands